src/vs/workbench/contrib/interactiveSession/browser/actions/interactiveSessionCodeblockActions.ts

```diff
@@ -118,12 +125,56 @@ export function registerInteractiveSessionCodeBlockActions() {
 				return;
 			}
 
-			const activeTextModel = textFileService.files.get(activeModel.uri);
+			// Check if model is editable, currently only support untitled and text file
+			const activeTextModel = textFileService.files.get(activeModel.uri) ?? textFileService.untitled.get(activeModel.uri);
 			if (!activeTextModel || activeTextModel.isReadonly()) {
 				return;
 			}
 
-			const activeSelection = activeEditorControl.getSelection() ?? new Range(activeModel.getLineCount(), 1, activeModel.getLineCount(), 1);
+			await this.handleTextEditor(accessor, activeEditorControl, activeModel, context);
+		}
+
+		private async handleNotebookEditor(accessor: ServicesAccessor, notebookEditor: INotebookEditor, context: IInteractiveSessionCodeBlockActionContext) {
+			if (!notebookEditor.hasModel()) {
+				return;
+			}
+
+			if (notebookEditor.isReadOnly) {
+				return;
+			}
+
+			if (notebookEditor.activeCodeEditor?.hasTextFocus()) {
+				const codeEditor = notebookEditor.activeCodeEditor as ICodeEditor;
+				const textModel = codeEditor.getModel();
+
+				if (textModel) {
+					return this.handleTextEditor(accessor, codeEditor, textModel, context);
+				}
+			}
+
+			const languageService = accessor.get(ILanguageService);
+			const interactiveSessionService = accessor.get(IInteractiveSessionService);
+
+			const focusRange = notebookEditor.getFocus();
+			const next = Math.max(focusRange.end - 1, 0);
+			insertCell(languageService, notebookEditor, next, CellKind.Code, 'below', context.code, true);
+
+			interactiveSessionService.notifyUserAction(<IInteractiveSessionUserActionEvent>{
+				providerId: context.element.providerId,
+				action: {
+					kind: 'insert',
+					responseId: context.element.providerResponseId,
+					codeBlockIndex: context.codeBlockIndex,
+					totalCharacters: context.code.length,
+				}
+			});
```

Comment:
Can this block be shared between the two cases?

Reply:
yep 👍